### PR TITLE
SQLite as demo database and simpler docker-compose setup

### PR DESCRIPTION
**Pull request by northwestwitch**
This is the first step to add real data to a demo instance

### This PR adds | fixes:
- Use a demo SQLite database when launching the demo app (docker-compose file or locally when no env vars are provided)
- Creates an additional docker-compose-mysql.yml file to show setup for a real database (MySQL) - that is basically the same docker-compose from before
- Renamed folder `dependencies` to `dbutil`

#### Test 1: run a demo with docker compose:
- `docker-compose up`

#### Test 2: run a demo connected to a MySQL database with docker compose:
- `docker-compose -f docker-compose-mysql.yml --env-file template.env up`

#### Test 3: run a demo from a local conda installation:
- `gunicorn --config gunicorn.conf.py src.chanjo2.main:app`

### Expected outcome:
- [x] All 3 options should work

### Review:
- [x] Code approved by HS
- [x] Tests executed by CR

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by northwestwitch** [comment]
### Test n.1 - Demo with SQLite:  🆗  
`docker-compose up`

image

image

### Test n.2 - Demo with MariaDB and custom params: 🆗 
`docker-compose -f docker-compose-mysql.yml --env-file template.env up` 

image

image

### Test n.2 - Launch demo from local conda env: 🆗 

image

File: README.md [hunk 1]
```diff
@@ -2,18 +2,30 @@
...
+The file named `Dockerfile` is a generic Docker file to run the application. Whenever the app is launched with the ENV param `DEMO` (check the settings present in file `docker-compose.yml`) it will create a test SQLite database to work with.
 
-To test the application connected to a database (MariaDB) instance, you could use the provided Docker-compose file (`docker-compose.yml`).
+To start the demo via docker-compose, run:
+
+```
+docker-compose up
+```
+
+The endpoints of the app will be now reachable from ant web browser: http://0.0.0.0:8000/docs or http://localhost:8000/docs
```
Comment (henrikstranneheim) [inline README.md:15]:
```suggestion
The endpoints of the app will be now reachable from an web browser: http://0.0.0.0:8000/docs or http://localhost:8000/docs
```

File: docker-compose-mysql.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+version: '3'
```
Comment (northwestwitch) [inline docker-compose-mysql.yml:1]: This is exactly the same file as docker-compose.yml, only renamed

File: docker-compose.yml [hunk 3]
```diff
@@ -1,46 +1,11 @@
 version: '3'
```
Comment (northwestwitch) [inline docker-compose.yml:1]: This is new

File: src/chanjo2/dbutil.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import os
+
+from sqlmodel import Session, create_engine
+
+engine = None
```
Comment (henrikstranneheim) [inline src/chanjo2/dbutil.py:5]: Would be nice with type hints wherever possible

File: src/chanjo2/dbutil.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+root_password = os.getenv("MYSQL_ROOT_PASSWORD")
+db_name = os.getenv("MYSQL_DATABASE_NAME")
+host_name = os.getenv("MYSQL_HOST_NAME")
+port_no = os.getenv("MYSQL_PORT")
+
+if os.getenv("DEMO") or not db_name:
+    mysql_url = "sqlite:///./chanjotest.db"
```
Comment (henrikstranneheim) [inline src/chanjo2/dbutil.py:13]: I think the database url can be imported as a constant to avoid hard-coding and reuse

**Review by henrikstranneheim — APPROVED** [review]
Looks great 👍

File: README.md (hunk 1 above)
Comment (northwestwitch) [reply README.md:15]: I think I meant any here :)

Comment (northwestwitch) [reply README.md:15]: 🐜

Comment (henrikstranneheim) [reply README.md:15]: Right 😄

File: src/chanjo2/dbutil.py (hunk 4 above)
Comment (northwestwitch) [reply src/chanjo2/dbutil.py:5]: It's not even needed, I'll remove the line!

**Pull request merged**